core/src/worker/workflow/mod.rs

```diff
@@ -1273,3 +1304,22 @@ impl LocalActivityRequestSink for LAReqSink {
         res
     }
 }
+
+/// Sorts jobs in an activation to be in the order lang expects:
+/// `patches -> signals -> other -> queries`
+fn sort_act_jobs(wfa: &mut WorkflowActivation) {
+    wfa.jobs.sort_by(|j1, j2| {
+        // Unwrapping is fine here since we'll never issue empty variants
+        let j1v = j1.variant.as_ref().unwrap();
+        let j2v = j2.variant.as_ref().unwrap();
+        if discriminant(j1v) == discriminant(j2v) {
+            return Ordering::Equal;
+        }
+        match (j1v, j2v) {
+            (workflow_activation_job::Variant::NotifyHasPatch(_), _) => Ordering::Less,
+            (workflow_activation_job::Variant::SignalWorkflow(_), _) => Ordering::Less,
+            (workflow_activation_job::Variant::QueryWorkflow(_), _) => Ordering::Greater,
+            _ => Ordering::Equal,
```

Comment:
Can you confirm just for my reading that this case should never be hit because of the `discriminant(j1v) == discriminant(j2v)` check above? (fine to remain for exhaustion, I just want to confirm that the match clause never compares equal elements)

Reply:
No, it never does.